internal/testprotos/desc_test3.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto2";
+
+option go_package = "github.com/jhump/protoreflect/internal/testprotos";
+
+import "google/protobuf/descriptor.proto";
+
+package testprotos;
+
+extend google.protobuf.MethodOptions {
+    optional Validator validator = 12345;
+}
+
+enum Action {
+    login = 0;
+    read = 1;
+    write = 2;
+}
+
+message Permission {
+    required Action action = 1;
```

Comment:
FYI: You very well may find more bugs in `protoparse` related to aggregate options that have required fields. This package is not trying to verify that required fields have been set. In order to mirror protoc behavior, it probably should.

Reply:
It's OK. I'm using this as an auxiliary tool to generate additional validation logic. When protoc is used to generate go source from these protos, I presume it will alarm if there is an error. My project is here: https://github.com/neighborly/grpc-auth